src/configuration.js

```diff
@@ -0,0 +1,252 @@
+        CONTAINER_URL: 'containerUrl',
+        USER_ID: 'userId',
+        PROXY_HOSTNAME: 'proxyHostname',
+        PROXY_PASSWORD: 'proxyPassword',
+        PROXY_STATUS_URL: 'proxyStatusUrl',
+        PROXY_PORT: 'proxyPort',
+        MAX_OPENED_STORAGES: 'maxOpenedStorages',
+
+        // not supported, use env vars directly:
+        // MEMORY_MBYTES: 'memoryMbytes',
+        // HEADLESS: 'headless',
+        // XVFB: 'xvfb',
+        // CHROME_EXECUTABLE_PATH: 'chromeExecutablePath',
+    };
+
+    static ENV_MAP_REVERSED = Object.entries(Configuration.ENV_MAP).reduce((obj, [key, value]) => {
+        obj[value] = key;
+        return obj;
+    }, {});
+
+    static BOOLEAN_VARS = ['localStorageEnableWalMode'];
+
+    static INTEGER_VARS = ['proxyPort', 'internalPort', 'memoryMbytes', 'containerPort'];
+
+    static DEFAULTS = {
+        defaultKeyValueStoreId: 'default',
+        defaultDatasetId: 'default',
+        defaultRequestQueueId: 'default',
+        inputKey: 'INPUT',
+        proxyHostname: 'proxy.apify.com',
+        apiBaseUrl: 'https://api.apify.com/v2',
+        proxyStatusUrl: 'http://proxy.apify.com',
+        proxyPort: 8000,
+        containerPort: 4321,
+        containerUrl: 'http://localhost:4321', // Must match `containerPort` above!
+        maxOpenedStorages: 1000,
+        metamorphAfterSleepMillis: 300e3,
+        persistStateIntervalMillis: 60e3, // This value is mentioned in jsdoc in `events.js`, if you update it here, update it there too.
+        localStorageEnableWalMode: true,
+    };
+
+    /**
+     * Creates new `Configuration` instance with provided options. Env vars will have precedence over those.
+     *
+     * @param {Record<string, number | string | boolean>} options
+     */
+    constructor(options = {}) {
+        this.options = options;
+
+        if (!this.get('localStorageDir') && !this.get('token')) {
+            const dir = join(process.cwd(), './apify_storage');
+            this.set('localStorageDir', dir);
+            // eslint-disable-next-line max-len
+            log.warning(`Neither ${ENV_VARS.LOCAL_STORAGE_DIR} nor ${ENV_VARS.TOKEN} environment variable is set, defaulting to ${ENV_VARS.LOCAL_STORAGE_DIR}="${dir}"`);
+        }
+    }
+
+    /**
+     * Returns configured value. First checks the environment variables, then provided configuration,
+     * fallbacks to the `defaultValue` argument if provided, otherwise uses the default value as described
+     * in the above section.
+     *
+     * @param {string} key
+     * @param {string | number | boolean} [defaultValue]
+     * @return {string | number | boolean}
+     */
+    get(key, defaultValue) {
+        // prefer env vars
+        const envKey = Configuration.ENV_MAP_REVERSED[key] ?? key;
+        const envValue = process.env[envKey] ?? process.env[ENV_VARS[envKey] ?? process.env[`APIFY_${envKey}`]];
+
+        if (envValue != null) {
+            return this._cast(key, envValue, true);
```

Comment:
I'm not a big fan of boolean args that alter behavior. Even after reading the JSDoc I still don't know what the `env` boolean actually does. Is it something like `isValueAnEnvVar` ? Here it doesn't matter much, but generally we prefer `fn(arg1, arg2, { isValueAnEnvVar: true })`

Reply:
I'd say using the indefinite article in a variable looks also extremly weird :] Will use options parameter, and try to pick more accurate name for that.

In fact this was rather a fail safe, we might not even need to do this conditionally. We could just check whether the value is a `string` instead of the `env` parameter.

Reply:
Right, we don't need that parameter at all, the method is used only to cast env values. Originally I had it for the regular ones from `options` too but realized that it is a user's responsibility to provide the right type, the casting is needed only for env vars.